FrontEnd/src/components/CompanyCard/CompanyCard.jsx

```diff
@@ -0,0 +1,209 @@
+import { Link } from 'react-router-dom';
+import { StarOutlined, StarFilled } from '@ant-design/icons';
+import { useState, useMemo } from 'react';
+import { useSWRConfig } from 'swr';
+import useSWRMutation from 'swr/mutation';
+import styles from './CompanyCard.module.css';
+import { useUser } from '../../hooks';
+import PropTypes from 'prop-types';
+import { Tooltip, Badge } from 'antd';
+
+export default function CompanyCard({ data, isAuthorized }) {
+  CompanyCard.propTypes = {
+    data: PropTypes.object,
+    isAuthorized: PropTypes.bool,
+  };
+  const [isSaved, setIsSaved] = useState(data && data.is_saved);
+  const { mutate } = useSWRConfig();
+  const profile = useMemo(() => {
+    return {
+      id: data?.id,
+      personId: data?.person,
+      name: data?.name,
+      region: data?.region,
+      categories: data?.categories,
+      activities: data?.activities,
+      logo: data?.logo_image,
+      banner: data?.banner_image,
+      founded: data?.founded,
+      isSaved: data?.is_saved,
+    };
+  }, [data]);
```

Comment:
is there any reason why we can use `data` directly? Does it make sense to provide better naming for `data`  like `profileData` or `profile`